fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java

```diff
@@ -742,9 +742,22 @@ private static LoanTransaction useOldTransactionIfApplicable(LoanTransaction old
         return null;
     }
 
-    protected void createNewTransaction(LoanTransaction oldTransaction, LoanTransaction newTransaction, TransactionCtx ctx) {
+    protected void createNewTransaction(final LoanTransaction oldTransaction, final LoanTransaction newTransaction,
+            final TransactionCtx ctx) {
         oldTransaction.updateExternalId(null);
         oldTransaction.getLoanChargesPaid().clear();
+
+        if (newTransaction.getTypeOf().isInterestRefund()) {
```

Comment:
I am not sure it is correct. We should create new relationship between interest refund and merchant issued refund. If i understand correctly here, you are trying to update the old one instead...